cursor.go

```diff
@@ -80,6 +81,10 @@ func (c *Cursor) Each(f func(*RowBatch) error) error {
 		c.ColumnsToSelect = []string{"*"}
 	}
 
+	if c.SelectFingerprint {
+		c.ColumnsToSelect = append(c.ColumnsToSelect, c.Table.RowMd5Query())
+	}
```

Comment:
There's two places in `Cursor` that check this flag.. This looks pretty leaky to me.

How about making the caller pass in the `ColumnsToSelect` option:

```go
func (c *Cursor) Each(columnsToSelect []string, f func(*RowBatch) error) error {
    ...
}
```

And since the caller has passed in the selection clause, they know what format of data to expect:
```go
// in data_iterator.go

err := cursor.Each(columnsIncludingMD5, func(batch *RowBatch) error {
  if c.SelectFingerprint {
    batchDataWithoutFingerprints []RowData
    fingerprints := make(map[uint64][]byte)

    for _, rowData := range batch.Values {
      // wrangle the resultset into `batchDataWithoutFingerprints` and `fingerprints`
    }

    batch = &RowBatch{
      values: batchDataWithoutFingerprints,
      fingerprints: fingerprints,
      pkIndex: batch.pkIndex,
      table: batch.Table,
    }
  }
  ...
}
```

Reply:
I tried this, but I still needed two ifs within DataIterator.Run

```
cursor := CursorConfig.NewCursor(....)
if dataiterator.SelectFingerprint { 
  cursor.ColumnsToSelect = ... // This is also harder to write here because Cursor.Each set a default value for ColumnsToSelect if not specified
}
cursor.Each(func(batch *RowBatch) {
  if dataiterator.SelectFingerprint {
    // Extract fingerprint from batch
  }
  batch = &RowBatch{....}
  ...
})
```

Reply:
> two ifs within DataIterator.Run

That sounds more reasonable than having it be within a general-purpose cursor IMO